script.js

```diff
@@ -1 +1,285 @@
+      if (cardName === '1') {
+        cardName = 'ace';
+      } else if (cardName === '11') {
+        cardName = 'jack';
+      } else if (cardName === '12') {
+        cardName = 'queen';
+      } else if (cardName === '13') {
+        cardName = 'king';
+      }
+
+      // make a single card object variable
+      const card = {
+        name: cardName,
+        suit: currentSuit,
+        rank: rankCounter,
+        color: 'red',
+      };
+
+      console.log(`rank: ${rankCounter}`);
+
+      // add the card to the deck
+      newDeck.push(card); // add double the cards to the deck
+      newDeck.push(card);
+    }
+  }
+
+  return newDeck;
+};
+const shortMessage = (displayMsgCounter) => {
+  winMessage.innerHTML = 'Match!';
+  if (displayMsgCounter === true) {
+    winMessage.innerText = '';
+  }
+  if (displayMsgCounter === false) {
+    winMessage.innerText = 'you win all cards';
+    console.log('test3');
+  }
+  document.body.appendChild(winMessage);
+};
+const squareClick = (cardElement, column, row) => {
+  console.log(cardElement);
+
+  console.log('FIRST CARD DOM ELEMENT', firstCard);
+
+  console.log('BOARD CLICKED CARD', board[column][row]);
+
+  const clickedCard = board[column][row];
+
+  // the user already clicked on this square
+  if (cardElement.innerText !== '') {
+    return;
+  }
+
+  // first turn
+  if (firstCard === null && canClick) {
+    console.log('first turn');
+    firstCard = clickedCard;
+    console.log(firstCard);
+    // turn this card over
+    cardElement.innerText = `${firstCard.name}\n${firstCard.suit}`;
+
+    // hold onto this for later when it may not match
+    firstCardElement = cardElement;
+
+    // second turn
+  } else {
+    console.log('second turn');
+    if (
+      clickedCard.name === firstCard.name
+        && clickedCard.suit === firstCard.suit
+    ) {
+      console.log('match');
+      let showMatch = '';
+      scoreIndex += 1;
+      updatingScore();
+      console.log(scoreIndex < 8);
+      console.log(scoreIndex === 1);
+      if (scoreIndex < 8) {
+        cardElement.innerText = `${clickedCard.name}\n${firstCard.suit}`;
+        cardElement.classList.add(clickedCard.colour);
+        console.log('test');
+        showMatch = shortMessage();
+        displayMsgCounter = true;
+        showMatch = setTimeout(() => {
+          shortMessage(displayMsgCounter);
+        }, 3000);
+        // turn this card over
+      }
+      if (scoreIndex === 8) {
+        console.log('test2');
+        cardElement.innerText = `${clickedCard.name}\n${firstCard.suit}`;
+        cardElement.classList.add(clickedCard.colour);
+        displayMsgCounter = false;
+        showMatch = shortMessage(displayMsgCounter);
+        displayMsgCounter = true;
+        showMatch = setTimeout(() => {
+          shortMessage(displayMsgCounter);
+        }, 50000);
+        console.log('1 point');
+        playerScore += 1;
+        scoreIndex = 0;
+      }
+    } else {
+      canClick = false;
```

Comment:
Nice way to prevent user from clicking! Can consider implementing this when there is a match as well, to prevent the user from clicking new cards before you show the `shortMessage`.